src/components/Board.js

```diff
@@ -5,22 +5,159 @@ import axios from 'axios';
 import './Board.css';
 import Card from './Card';
 import NewCardForm from './NewCardForm';
-import CARD_DATA from '../data/card-data.json';
+// import CARD_DATA from '../data/card-data.json';
 
 class Board extends Component {
   constructor() {
     super();
 
     this.state = {
       cards: [],
+      errors: [],
+      ids: [],
     };
+  };
+
+  componentDidMount() {
+    const GET_ALL_CARDS_URL = "https://inspiration-board.herokuapp.com/boards/redPanda/cards";
+
+    axios.get(GET_ALL_CARDS_URL)
+    .then((response) => {
+      this.setState({
+        errors: []
+      });
+
+      let updatedIDsAfterGet = response.data.map((card) => {
+        // console.log(card.card.id);
+        return card.card.id
+      })
+
+      this.setState({
+        cards: response.data,
+        ids: updatedIDsAfterGet,
+      });
+    })
+    .catch((error) => {
+      this.setState({
+        errors: error.response.data.errors.text
+      });
+    });
   }
 
+  displayErrorsToUser = () => {
+    // console.log("I am in displayErrorsToUser");
+    // console.log(this.state.errors);
+    if (this.state.errors.length >= 1) {
+      let catchErrors = this.state.errors.map(() => {
+        return <li>{this.state.errors}</li>
+      });
+
+      return <ul>{catchErrors}</ul>;
+    }
+    else {
+      return;
+    };
+  };
+
+
+
+
+  addCardintoCardCollection = (newCardInfo) => {
+    const POST_CARD_TO_BOARD_URL = "https://inspiration-board.herokuapp.com/boards/redPanda/cards";
+
+     axios.post(POST_CARD_TO_BOARD_URL, newCardInfo)
+     .then((response) => {
+       this.setState({
+         errors: []
+       });
+
+       // console.log("In the success response for the post", response);
+       let updatedData = this.state.cards;
+       updatedData.push({card: response.data.card});
+
+       let updatedIDsAfterPost = this.state.ids;
+       updatedIDsAfterPost.push(response.data.card.id);
+       // console.log(updatedIDsAfterPost);
+
+       this.setState({
+         cards: updatedData,
+         ids: updatedIDsAfterPost,
+
+       });
+     })
+     .catch((error) => {
+       this.setState({
+         errors: error.response.data.errors.text
+       });
+     });
+  };
+
+  deleteCard = (cardNumber) => {
+    // console.log("I am in deleteCard");
+    // console.log(cardNumber);
+    let DELETE_CARD_URL = `https://inspiration-board.herokuapp.com/cards/${cardNumber}`
+    axios.delete(DELETE_CARD_URL)
+      .then((response) => {
```

Comment:
I like that you've kept all the API interaction logic in one place - the callbacks are a little more complex, but I would say it makes the app as a whole much easier to comprehend. Whether or not you intended it, this is a great example of the container component pattern well-applied.